### ISSUE TITLE
fix: npm package "urllib": invalid type: map, expected a string

### ISSUE THREAD

**Issue by iugo** [denoland/deno#23862]
Version: Deno 1.43.4

There is a file named test.ts with the following content:

```ts
import urllib from 'npm:urllib@2.41.0';

console.log(urllib);
```

An error occurred when executing the command `deno check -r test.ts`:

```
error: Error getting response at https://registry.npmjs.org/urllib for package "urllib": invalid type: map, expected a string at line 1 column 24565
    at file:///Users/.../test.ts:1:20
```

This issue was not present in Deno 1.43.3 according to the actual test.

**Comment by hugojosefson** [comment]
:raising_hand: The same with `npm:joi`:

```ts
import joi from "npm:joi@17.13.1";

console.log(joi);
```

Works with Deno version 1.43.3.

Fails with Deno version 1.43.4:

```
error: Error getting response at https://registry.npmjs.org/joi for package "joi": invalid type: map, expected a string at line 1 column 18384
    at file:///test.ts:1:17
```

The url https://registry.npmjs.org/joi, does indeed return json that, at the position given in the error message, contains a run-script value that isn't a string, but a json object. It was present in the ancient `joi@0.1.0`, removed in the almost as ancient `joi@0.2.1`.

- https://github.com/hapijs/joi/blob/v0.1.0/package.json#L30
- https://github.com/hapijs/joi/blob/v0.2.1/package.json

### :ginger_root: Root cause?

I suspect this newly found insensitivity to incorrect `package.json` run-scripts, was brought in by this entry in the [deno 1.43.4 changelog](https://github.com/denoland/deno/releases/tag/v1.43.4):

> fix(npm): handle null fields in npm registry JSON (https://github.com/denoland/deno/pull/23785)

Following that update, leads to this commit in `deno_npm`:

https://github.com/denoland/deno_npm/compare/0.20.0...0.20.1#diff-d5b93d4a7caa06c359c97e5e8528a0b6c636621edba60a960f3f5afeed0c97a8R171

That's how far my understanding goes! I'm not familiar enough with serde in rust to know what the previous default behaviour was, that was apparently overridden here. In general though, for npm compatibility, we might want to be more lenient to parts of json from older versions of modules that we don't even need to use?

**Comment by bartlomieju** [comment]
CC @dsherret can you take a look? Seems like a regression.

**Comment by dsherret** [comment]
I think this was caused by https://github.com/denoland/deno_npm/pull/47

**Comment by bartlomieju** [comment]
It looks so, here's the problematic bit:
```
"scripts": {
    "test": "make test && make test-cov",
    "blanket": {
        "pattern": "//^((?!/node_modules/)(?!/test/).)*$/ig",
        "onlyCwd": true,
        "data-cover-flags": {
            "branchTracking": true
        }
    },
    "travis-cov": {
        "threshold": 100
    }
},
```

**Comment by dsherret** [comment]
This is now fixed in Deno 1.43.5

**Comment by hugojosefson** [comment]
Thank you so much @dsherret and @bartlomieju!
